mlx/backend/metal/kernels/utils.h

```diff
@@ -3,7 +3,11 @@
 #pragma once
 
 #include <metal_math>
-#include "mlx/backend/metal/kernels/bf16.h"
+
+// bf16.h is conditionally included based on the metal version
+#include "bf16.h"
```

Comment:
Instead of using the preprocessor include the correct bf16 file in CMake here.

Reply:
I know what you mean but only because I read the PR. Perhaps "The appropriate bf16.h will be included based on the metal version" or "A different bf16.h will be included based on the metal version"?